pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1258,6 +1258,7 @@ void messageReceived(CommandMessage cmdMessage, ByteBuf headersAndPayload, Clien
         if (!verifyChecksum(headersAndPayload, messageId)) {
             // discard message with checksum error
             discardCorruptedMessage(messageId, cnx, ValidationError.ChecksumMismatch);
+            headersAndPayload.release();
```

Comment:
The call path here would be 
- org.apache.pulsar.common.protocol.PulsarDecoder#channelRead
- org.apache.pulsar.client.impl.ClientCnx#handleMessage
- org.apache.pulsar.client.impl.ConsumerImpl#messageReceived

But the `headersAndPayload` is released in the `finally` block in `channelRead`

Reply:
@Jason918 Thanks, I have forgot that `finally` block.